app/scripts/migrations/127.ts

```diff
@@ -0,0 +1,265 @@
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by
+ * controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  if (
+    !hasProperty(state, 'PermissionController') ||
+    !isObject(state.PermissionController) ||
+    !hasProperty(state, 'NetworkController') ||
+    !isObject(state.NetworkController) ||
+    !hasProperty(state, 'SelectedNetworkController') ||
+    !isObject(state.SelectedNetworkController)
+  ) {
+    return state;
+  }
+  const {
+    PermissionController: { subjects },
+    NetworkController: { selectedNetworkClientId, networkConfigurations },
+    SelectedNetworkController: { domains },
+  } = state;
+  if (
+    !isObject(subjects) ||
+    !selectedNetworkClientId ||
+    typeof selectedNetworkClientId !== 'string' ||
+    !isObject(networkConfigurations) ||
+    !isObject(domains)
+  ) {
+    return state;
+  }
+
+  const getChainIdForNetworkClientId = (networkClientId: string) => {
+    const networkConfiguration =
+      (networkConfigurations[networkClientId] as { chainId: Hex }) ??
+      BUILT_IN_NETWORKS[
+        networkClientId as unknown as keyof typeof BUILT_IN_NETWORKS
+      ];
+    return networkConfiguration?.chainId;
+  };
+
+  const currentChainId = getChainIdForNetworkClientId(selectedNetworkClientId);
+  if (!currentChainId || typeof currentChainId !== 'string') {
+    return state;
+  }
+
+  for (const [origin, subject] of Object.entries(subjects)) {
+    if (!isObject(subject)) {
+      return state;
+    }
+
+    const { permissions } = subject as {
+      permissions: Record<string, PermissionConstraint>;
+    };
+    if (!isObject(permissions)) {
+      return state;
+    }
+
+    let basePermission = {};
+
+    let ethAccounts: string[] = [];
+    if (
+      isObject(permissions[PermissionNames.eth_accounts]) &&
+      Array.isArray(permissions[PermissionNames.eth_accounts].caveats)
+    ) {
+      ethAccounts =
+        (permissions[PermissionNames.eth_accounts].caveats?.[0]
+          ?.value as string[]) ?? [];
+      basePermission = permissions[PermissionNames.eth_accounts];
+    }
+    delete permissions[PermissionNames.eth_accounts];
+
+    let chainIds: string[] = [];
+    if (
+      isObject(permissions[PermissionNames.permittedChains]) &&
+      Array.isArray(permissions[PermissionNames.permittedChains].caveats)
+    ) {
+      chainIds =
+        (permissions[PermissionNames.permittedChains].caveats?.[0]
+          ?.value as string[]) ?? [];
+      basePermission ??= permissions[PermissionNames.permittedChains];
```

Comment:
I don't believe this will ever obtain, since `basePermissions` will be an empty object or an `eth_accounts` permission object

Reply:
We could do
```suggestion
      basePermission ||= permissions[PermissionNames.permittedChains];
```
But I don't think we can use ES12 features yet

Reply:
```suggestion
      if(Object.keys(basePermission).length === 0) {
      basePermission = permissions[PermissionNames.permittedChains];
      }
```

Reply:
fixed [here](https://github.com/MetaMask/metamask-extension/pull/26483/commits/bd62ea82c6da81ef1af084a4a4c0f0b128d04144) by making `basePermissions` initial assignment undefined instead of empty object